File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function (nums) {
    let map = new Map()
    nums.forEach((value) => {
        if (map.has(value)) {
            map.set(value, map.get(value) + 1)
        }
        else {
            map.set(value, 1)
        }
    })
    for (let [val, type] of map) {
        if (type > 1) return val
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        map = {}
        for i in range(len(nums)):
            if nums[i] in map:
                map[nums[i]] = map[nums[i]] + 1
            else:
                map[nums[i]] = 1
        for value,index in map.items():
            if index > 1:
                return value